### PR TITLE
New register extensions

### PR DESCRIPTION
**Pull request by ericpre**
Some suggestions and tweaks for the dev guide.

### CODE REVIEW

File: doc/dev_guide/writing_extensions.rst [hunk 1]
```diff
@@ -230,26 +230,31 @@ Equivalently, to add a new component 2D:
...
 
-Toolkeys are functions to which it is possible to associate widgets. Extension
-can declare new toolkeys and widgets. For example, the `hyperspy-gui-traitsui
-<https://github.com/hyperspy/hyperspy_gui_traitsui>`_ and
-`hyperspy-gui-ipywidgets
-<https://github.com/hyperspy/hyperspy_gui_ipywidgets>`_ provide widgets for
-toolkeys declared in HyperSpy.
+To generate GUIs of specific method and functions, HyperSpy use widgets and toolkeys:
+
+* *widgets* are ipywidgets or traitsui objects, which can generate GUIs,
```
Comment (francisco-dlp) [inline doc/dev_guide/writing_extensions.rst]: That's not entirely true. In principle you could write a new widget "provider" in this way. Actually, that's what ``hyperspy-gui-traitsui`` and ``hyperspy-gui-ipywidgets`` do

File: doc/dev_guide/writing_extensions.rst [hunk 2]
```diff
@@ -230,26 +230,31 @@ Equivalently, to add a new component 2D:
...
 
 Registering toolkeys
 ^^^^^^^^^^^^^^^^^^^^
+To register a new toolkey:
 
-Typically new toolkeys are declared using the
-:py:func:`hyperspy.ui_registry.add_gui_method` HyperSpy decorator.
-To register a new toolkey that you have declared in your package, add it to
-the ``hyperspy_extension.yaml`` file as in the following example:
+1. declare a new toolkey by adding the :py:func:`hyperspy.ui_registry.add_gui_method`
```
Comment (francisco-dlp) [inline doc/dev_guide/writing_extensions.rst]: I would add "e.g." in front of ``add_gui_method`` because there are other ways to do it.

**Review by francisco-dlp — COMMENTED** [review]
Except for the small comments above, LGTM!

File: doc/dev_guide/writing_extensions.rst (hunk 1 above)
Comment (ericpre) [reply doc/dev_guide/writing_extensions.rst]: Indeed!

File: doc/dev_guide/writing_extensions.rst (hunk 2 above)
Comment (ericpre) [reply doc/dev_guide/writing_extensions.rst]: Shall we also mention the `get_gui` function?

Comment (francisco-dlp) [reply doc/dev_guide/writing_extensions.rst]: I think that most developers will simply read the code to find it out, but do as you feel is best.

**Comment by francisco-dlp** [comment]
LGTM. Let me know if I should wait for extra commits or merge.

**Comment by ericpre** [comment]
No, that's good enough.

**Pull request merged**